### PR TITLE
GCW- 2421- Add total request count to order's contact summary

### PR DESCRIPTION
**Pull request by abulsayyad123**
Hi Team,
This PR is for ticket GCW-2421- Adding total request counts. Since, we have common routes for fetching orders data in nested routes. Which will results in unnecessary request i.e orders_count, for data which is not required in other pages. So, I have added separate ajax request for orders_count on orders_summary page instead of adding it in users or orders payload.
Please review the PR and let me know if any change is required.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -332,7 +332,7 @@ def stockit_abilities
   def user_abilities
     can [:current_user_profile], User
     can [:show, :update], User, id: @user_id
-    can [:index, :show, :update, :recent_users], User if can_read_or_modify_user?
+    can [:index, :show, :update, :recent_users, :orders_count], User if can_read_or_modify_user?
```
Comment (namrataukirde) [inline app/models/ability.rb]: Why we need to add this with `can_read_or_modify_user?` permission?

File: app/models/order.rb [hunk 2]
```diff
@@ -96,6 +96,11 @@ def can_dispatch_item?
     ORDER_UNPROCESSED_STATES.include?(state)
   end
 
+  def self.counts_for(created_by_id)
+    order_state_counts = group(:state).where(created_by_id: created_by_id).count
```
Comment (namrataukirde) [inline app/models/order.rb]: How about  `where(state IN (?), valid_states_array).distinct.group(:state).where(:created_by_id).count` ?

Could please check which is better one and let us know?

Comment (abulsayyad123) [reply app/models/order.rb]: Sure, will try it out.

Comment (abulsayyad123) [reply app/models/order.rb]: I benchmarked performance of both queries.  The second query is little slow on an average of 2-3 ms.

Comment (abulsayyad123) [reply app/models/order.rb]: Added refactored query.

**Pull request merged**